packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -74,16 +81,38 @@ class TextPainter {
   /// How the text should be aligned horizontally.
   ///
   /// After this is set, you must call [layout] before the next call to [paint].
+  ///
+  /// The [textAlign] property must not be null. It defaults to [TextAlign.start].
   TextAlign get textAlign => _textAlign;
   TextAlign _textAlign;
   set textAlign(TextAlign value) {
+    assert(value != null);
     if (_textAlign == value)
       return;
     _textAlign = value;
     _paragraph = null;
     _needsLayout = true;
   }
 
+  /// The default directionality of the text.
+  ///
+  /// This controls how direction-agnostic values of [textAlign], such as
+  /// [TextAlign.start], are resolved.
```

Comment:
direction-agnostic <--- this opposite of this, right?  It might be clearer just to list the two values that are controlled.

Reply:
listed the three values explicitly